src/commands/usages/index.ts

```diff
@@ -101,13 +105,21 @@ export default class Usages extends Base {
 
         const variableAliases = getVariableAliases(flags, this.repoConfig)
         
-        const matchesByVariable = this.getMatchesByVariable(matchesBySdk, variableAliases)
+        const usages = this.getMatchesByVariable(matchesBySdk, variableAliases)
+        if (flags['only-unused']) {
+            const variables = await fetchAllVariables(this.authToken, this.projectKey)
+            Object.keys(usages).forEach((variableKey) => {
+                if (variables.find((v) => v.key === variableKey)) {
```

Comment:
this could potentially be iterating a large array many times, we could convert it to a map by key so that we can do a constant lookup